internal/provisioner/teleport.go

```diff
@@ -23,11 +23,12 @@ type teleport struct {
 	kops           *kops.Cmd
 	cluster        *model.Cluster
 	logger         log.FieldLogger
-	desiredVersion string
-	actualVersion  string
+	desiredVersion model.UtilityVersion
+	actualVersion  model.UtilityVersion
+	valuesPath     string
```

Comment:
I think this field is not used and probably not needed as the `desiredVersion` should have the values path.

Reply:
It must be used if the compiler didn't complain but you're right there's likely a simplification that can occur here. I'll have a look.

Reply:
Good catch @Szymongib thank you